tests/python/contrib/test_ethosn/test_networks.py

```diff
@@ -0,0 +1,143 @@
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Ethos-N integration end-to-end network tests"""
+
+import pytest
+pytest.importorskip('tflite')
+pytest.importorskip('tensorflow')
+
+from tvm import relay
+from tvm.relay.op.contrib.ethosn import ethosn_available, Available
+from tvm.contrib import download
+import tvm.relay.testing.tf as tf_testing
+import tflite.Model
+from . import infrastructure as tei
+
+
+def _get_tflite_model(tflite_model_path, inputs_dict, dtype):
+    with open(tflite_model_path, 'rb') as f:
+        tflite_model_buffer = f.read()
+
+    try:
+        tflite_model = tflite.Model.Model.GetRootAsModel(tflite_model_buffer, 0)
+    except AttributeError:
+        tflite_model = tflite.Model.GetRootAsModel(tflite_model_buffer, 0)
+    shape_dict = {}
+    dtype_dict = {}
+    for input in inputs_dict:
+        input_shape = inputs_dict[input]
+        shape_dict[input] = input_shape
+        dtype_dict[input] = dtype
+
+    return relay.frontend.from_tflite(
+        tflite_model,
+        shape_dict=shape_dict,
+        dtype_dict=dtype_dict,
+    )
+
+
+def _test_image_network(model_url, model_sub_path, input_dict, compile_hash, output_count, run=True, host_ops=0, npu_partitions=1):
+    if not ethosn_available():
+        return
+
+    def get_model():
+        if model_url[-3:] in ("tgz", "zip"):
+            model_path = tf_testing.get_workload_official(
+                model_url,
+                model_sub_path,
+            )
+        else:
+            model_path = download.download_testdata(
+                model_url,
+                model_sub_path,
+            )
+        return _get_tflite_model(model_path, input_dict, 'uint8')
+
+    outputs = []
+    inputs = {}
+    for input_name in input_dict:
+        input_shape = input_dict[input_name]
+        inputs[input_name] = tei.get_real_image(input_shape[1], input_shape[2])
+
+    for npu in [False, True]:
+        mod, params = get_model()
+        graph, lib, params = tei.build(mod, params, npu=npu, expected_host_ops=host_ops, npu_partitions=npu_partitions)
+        if npu:
+            tei.assert_lib_hash(lib, compile_hash)
+        if run:
+            outputs.append(tei.run(graph, lib, params, inputs, output_count, npu=npu))
+
+    if run:
+        tei.verify(outputs, 1, verify_saturation=False)
+
+
+def test_mobilenet_v1():
+    hw = ethosn_available()
+    _test_image_network(
+        model_url="https://storage.googleapis.com/download.tensorflow.org/" \
+                  "models/mobilenet_v1_2018_08_02/mobilenet_v1_1.0_224_quant.tgz",
+        model_sub_path="mobilenet_v1_1.0_224_quant.tflite",
+        input_dict={"input": (1, 224, 224, 3)},
+        compile_hash="81637c89339201a07dc96e3b5dbf836a",
```

Comment:
How this hash comes from? In case someone in the future changes Ethos-N codegen, how can this be updated?

Reply:
This is created by hashing the library output we get when compiling the model. We create the hash at a 'known good' point in time (i.e. when we run the HW tests internally and see everything is correct). The thinking here is that this provides a quick way to test whether changes in TVM have resulted in a behavioural change to our codegen.

We will remain responsible for updating these hashes while no public hardware is available. Should it become available, we'll no longer need the hashes as we can just test execution correctness directly.

Reply:
I see. Could you comment this on the code with TODO? Thanks.

Reply:
I've added in a comment rather than a TODO because I don't think this represents work that is yet to be done, but rather maintenance procedure. It explains the purpose of the hash and who to contact (myself and @Leo-arm ) should the test fail.